examples/classic/distributed/TwoPhaseCommit/two_phase_commit.qnt

```diff
@@ -0,0 +1,162 @@
+// -*- mode: Bluespec; -*-
+
+/* This specification stems from the specification of the two phase protocol
+   by Leslie Lamport:
+
+   https://github.com/tlaplus/Examples/blob/master/specifications/transaction_commit/TwoPhase.tla */
+
+module two_phase_commit {
+  type ResourceManagerState = Working | Prepared | Committed | Aborted
+  var resourceManagerStates: str -> ResourceManagerState
+
+  type TransactionManagerState = Init | Done
+  var transactionManagerState: TransactionManagerState
+
+  /// The set of resource managers from which the transaction manager has
+  /// received RMPrepared messages
+  var preparedResourceManagers: Set[str]
+
+  type Message =
+    | Abort
+    | Commit
+    | RMPrepared(str)
+    | RMCommitted(str)
+
+  var messages: Set[Message]
+
+  const resourceManagers: Set[str]
```

Comment:
It's not really "move this", because I wasn't trying to translate this directly from the TLA spec, so it was not intentionally changed. I actually didn't thought about it much, and sure, it makes more sense at the start of the spec. Will change.